backend/experiment/actions/score.py

```diff
@@ -31,7 +46,9 @@ def __init__(self, session, title: str = None, score=None, score_message=None, c
             get_rounds_passed=session.get_rounds_passed(),
             total_rounds=self.session.block.rounds
         )
-        self.score = score or session.last_score()
+        self.score = (
+            score if score else result.score if result else session.last_score()
```

Comment:
This line confuses me a bit. Can you maybe split it up in two lines? Also, can we assume that `result` will always be defined if there is no `score` and `result.score` won't throw an error?

Reply:
There is another check `if result`, so both can be undefined. It's a fallback to a fallback, really. I'll write proper if / else clauses which might make this more clear.